src/TRestTrackLinearizationProcess.cxx

```diff
@@ -0,0 +1,230 @@
+/*************************************************************************
+ * This file is part of the REST software framework.                     *
+ *                                                                       *
+ * Copyright (C) 2016 GIFNA/TREX (University of Zaragoza)                *
+ * For more information see http://gifna.unizar.es/trex                  *
+ *                                                                       *
+ * REST is free software: you can redistribute it and/or modify          *
+ * it under the terms of the GNU General Public License as published by  *
+ * the Free Software Foundation, either version 3 of the License, or     *
+ * (at your option) any later version.                                   *
+ *                                                                       *
+ * REST is distributed in the hope that it will be useful,               *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the          *
+ * GNU General Public License for more details.                          *
+ *                                                                       *
+ * You should have a copy of the GNU General Public License along with   *
+ * REST in $REST_PATH/LICENSE.                                           *
+ * If not, see http://www.gnu.org/licenses/.                             *
+ * For the list of contributors see $REST_PATH/CREDITS.                  *
+ *************************************************************************/
+
+//////////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////
+/// The TRestTrackLinearization process is meant to analyze linear tracks,
+/// such as alpha tracks. It should retreive the information after performing
+/// TRestTrackLinearization process, not implemented for XYZ tracks.
```

Comment:
`GetHitsProjection` was implemented time ago to reconstruct the profile a physical track event along the track. The track could be a complex electron track event (no need to be linear). I understand this process is doing now something else, I could enter in the details of how this is being used here, but I dont see how this is required to perform the fitting to the raw hits.

The process producing a track profile could be probably named `TRestTrackProfileProcess`, generating a std::vector or a TGraph with N points where the hits energy have been projected.

Reply:
Sorry, I don't understand this comment. Do you want to rename `TRestTrackLinearizationProcess` to `TRestTrackProfileProcess`?

Perhaps the naming of `GetHitsProjection` function is not adequate? I can rename the function. This function is actually performing the track linearization.

Or you want to split the process in two different processes one doing the linear fit or profile and another process doing the average of the closest hits? Also, I don't understand why do you want to store the results as an `std::vector` or `TGraph` since I believe is better to store as `TRestVolumeHits`

Reply:
We should discuss this online